### PR TITLE
Fixes

### LINKED ISSUE #301: Build fail for newer rack builds.
OS (if applicable): 10.10.5
Version (or "dev" if compiling from source):dev 

Cannot build from source. Gets following error during make:
In file included from src/core/MidiCCToCV.cpp:1:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/list:174:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/memory:1617:20: error: 
object expression of non-scalar type 'bool [3]' cannot be used in a
pseudo-destructor expression

Not sure if this is an xcode issue or what, but I can't build any of the new ones with the sample rate change.

### PR DESCRIPTION
**Pull request by bontric**
* Fix sync if ccNum is changed by modifying the text
* Fix #301 
* Clean up MidiIO and Midi modules implementation
* Fix misuse of static variables
* Add Smoothing for midi CC value transitions

### CODE REVIEW

File: src/core/MidiClockToCV.cpp [hunk 1]
```diff
@@ -34,6 +34,20 @@ struct MIDIClockToCVInterface : MidiIO, Module {
...
 	int c_bar = 0;
 
+	/* Note this is in relation to the Midi clock's Tick (6x per 16th note).
+	 * Therefore, e.g. the 2:3 is calculated:
+	 *
+	 * 24 (Ticks per quarter note) * 2 / 3 = 16
+	 *
+	 * Implying that every 16 midi clock ticks we need to send a pulse
+	 * */
+	int ratios[] = {6, 8, 12, 16, 24, 32, 48, 96, 192};
```
Comment (DrChrisBoucher) [inline src/core/MidiClockToCV.cpp]: This doesn't build for me (Windows, gcc 7.2.0):

`error: flexible array member 'MIDIClockToCVInterface::ratios' not at end of 'struct MIDIClockToCVInterface'`

Fixing the array size with `ratios[9]` works.

File: src/core/MidiClockToCV.cpp [hunk 2]
```diff
@@ -34,6 +34,20 @@ struct MIDIClockToCVInterface : MidiIO, Module {
...
+	int numratios = sizeof(ratios) / sizeof(*ratios);
+
+	/*
+	 * Length of clock pulse
+	 */
+	float pulseTime = 0.05;
```
Comment (DrChrisBoucher) [inline src/core/MidiClockToCV.cpp]: `pulseTime` could be made `const`

File: src/core/MidiIO.cpp [hunk 3]
```diff
@@ -50,7 +50,7 @@ void MidiIO::baseFromJson(json_t *rootJ) {
 
 std::vector<std::string> MidiIO::getDevices() {
 	/* Note: we could also use an existing interface if one exists */
-	static RtMidiIn *m = new RtMidiIn();
```
Comment (DrChrisBoucher) [inline src/core/MidiIO.cpp]: `m` is never deleted, so it leaks one everytime `getDevices()` is called.
It leaked when it was static too, but only one copy.
Could either delete `m` at the end of the function, or just use `RtMidiIn m;`

File: src/core/MidiClockToCV.cpp (hunk 2 above)
Comment (AndrewBelt) [reply src/core/MidiClockToCV.cpp]: who cares??

File: src/core/MidiIO.cpp (hunk 3 above)
Comment (AndrewBelt) [reply src/core/MidiIO.cpp]: Or rather
```cpp
static RtMidiIn *m = NULL;
if (!m)
	m = new RtMidiIn();
```
to be explicit.

Comment (DrChrisBoucher) [reply src/core/MidiIO.cpp]: Still leaks, but only one copy over the life of the program. Not a big deal but can be annoying if you start using memory analysis tools

Comment (bontric) [reply src/core/MidiIO.cpp]: Yeah both work but I'd go with the `RtMidiIn m` solution. It looks clean and the function is not called very often so this is not relevant for performance.

Comment (bontric) [reply src/core/MidiIO.cpp]: I added a proper check to see whether there already is a device and use that.

**Comment by AndrewBelt** [comment]
Is this complete or do you plan on adding more features? Some time before Nov 1, could you do a feature freeze and clean up the code the best you can, and then I'll go and review everything?

**Comment by bontric** [comment]
I'd rather focus on testing and cleaning up the code for now. There are some minor issues I encounter here and there so I think it's a better idea to play around a little and see what needs fixing.

One thing: Do you prefer any coding style/ characters per column? I don't particularly care but would like to run a formatter on the code.

**Comment by AndrewBelt** [comment]
Don't worry about formatting, I'll handle that if needed.

**Comment by bontric** [comment]
Okay, I'll notify you once this is done.

**Comment by bontric** [comment]
@AndrewBelt After another few hours of testing, formatting and fixing I'm pretty confident that this is done.

As you probably read in #81 I decided to add smoothing to midi cc (and now also pitch bend and mod wheel). For now I decided to not implement this for after touch since I don't own an after touch capable controller so I don't know how it would turn out.

And of course, if something needs further explanation feel free to ask..

**Comment by bontric** [comment]
I messed up the Trigger to CV, I'm working on a fix but should be done soon

**Comment by bontric** [comment]
@AndrewBelt I'm working on an update to make the modules use  the new event handling (8e251c0ddbfc1f4a2e03a6123d8eda42218b76fe)

**Comment by AndrewBelt** [comment]
Okay, I'll merge soon, I promise. Let me know and I'll do that pull+review.

**Comment by bontric** [comment]
@AndrewBelt It's done :)
> Okay, I'll merge soon, I promise.

Don't worry, there is no rush. You are busy enough as it is ;)

**Comment by AndrewBelt** [comment]
Double check that the merge didn't break anything. I had to add some code to make the CCTextFields work with left click (enter with keyboard) and right click (MIDI learn).

**Comment by bontric** [comment]
> I had to add some code to make the CCTextFields work with left click (enter with keyboard) and right click (MIDI learn).

Looks good to me but I'll try to do some more testing before v0.5 (When do you want to release 0.5?)

**Pull request merged**